core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1911,7 +1911,9 @@ public View getPrimaryView() {
      }
 
     public void setPrimaryView(@Nonnull View v) {
-        this.primaryView = v.getViewName();
+        if (hasPermission(ADMINISTER)) {
```

Comment:
> Why is this not CONFIGURE?

You would not be able to create Views (or even see them without other Permissions) so although this appears strange, I think it is technically correct.  However yes - this should `checkPermission(ADMINISTER)` so that the correct Exception is thrown.

Reply:
changed to checkPermission.

Reply:
Looking at the calling code I think this is not quite correct.
`GlobalDefaultViewConfiguration` should not try setting this if it doesn't have the permissions as this will always be sent in a save() IIUC causing saves of global configuration to fail?

Perhaps `GlobalConfiguration` needs to be enhanced to have `GlobalSensitiveConfiguration` or the like that a configure user will not be able to adjust?